composer/core/state.py

```diff
@@ -1278,14 +1278,14 @@ def _load_dataset_state(self, obj: dict[str, Any]) -> None:
         Args:
             obj (dict[str, Any]): The state to load.
         """
-        self.dataset_state = obj
-
         dataset = self._dataset_of(self.train_dataloader)
         if hasattr(dataset, 'load_state_dict'):
             dataset.load_state_dict(obj['train'])  # pyright: ignore
             obj['train'] = None
             self.dataset_resumption['train'] = True
 
+        self.dataset_state = obj
```

Comment:
I don't think anything changes if we move this line down here, since `obj['train']` will be `None` regardless (changing keys of the same dict, and it's not copied).

Reply:
Also, just so I'm understanding correctly -- we set `obj['train']` to `None` after loading in the dataset state so that we know that we've already loaded in the dataset state, right?

Reply:
Yes to understanding.

yes doesn't change anything, I think just easier to follow logic